core/src/test/java/org/apache/iceberg/rest/responses/TestFetchScanTasksResponseParser.java

```diff
@@ -0,0 +1,180 @@
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid fetch scan tasks response: null or empty");
+  }
+
+  @Test
+  public void missingRequiredField() {
+    String missingRequiredFieldJson = "{\"x\": \"val\"}";
+    assertThatThrownBy(() -> FetchScanTasksResponseParser.fromJson(missingRequiredFieldJson))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid response: planTasks and fileScanTask cannot both be null");
+  }
+
+  @Test
+  public void roundTripSerdeWithPlanTasks() {
+    String expectedJson = "{\"plan-tasks\":[\"task1\",\"task2\"]}";
+    String json =
+        FetchScanTasksResponseParser.toJson(
+            FetchScanTasksResponse.builder().withPlanTasks(List.of("task1", "task2")).build());
+    assertThat(json).isEqualTo(expectedJson);
+
+    FetchScanTasksResponse fromResponse = FetchScanTasksResponseParser.fromJson(json);
+
+    // can't do an equality comparison on PlanTableScanRequest because we don't implement
+    // equals/hashcode
+    assertThat(FetchScanTasksResponseParser.toJson(fromResponse, false)).isEqualTo(expectedJson);
+  }
+
+  @Test
+  public void roundTripSerdeWithDeleteFilesNoFileScanTasksPresent() {
+    assertThatThrownBy(
+            () ->
+                FetchScanTasksResponse.builder()
+                    .withPlanTasks(List.of("task1", "task2"))
+                    .withDeleteFiles(List.of(FILE_A_DELETES))
+                    .build())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage(
+            "Invalid response: deleteFiles should only be returned with fileScanTasks that reference them");
+
+    String invalidJson =
+        "{\"plan-tasks\":[\"task1\",\"task2\"],"
+            + "\"delete-files\":[{\"spec-id\":0,\"content\":\"POSITION_DELETES\","
+            + "\"file-path\":\"/path/to/data-a-deletes.parquet\",\"file-format\":\"PARQUET\","
+            + "\"partition\":{\"1000\":0},\"file-size-in-bytes\":10,\"record-count\":1}]"
+            + "}";
+
+    assertThatThrownBy(() -> FetchScanTasksResponseParser.fromJson(invalidJson))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage(
+            "Invalid response: deleteFiles should only be returned with fileScanTasks that reference them");
+  }
+
+  @Test
+  public void roundTripSerdeWithFileScanTasks() {
+    ResidualEvaluator residualEvaluator =
+        ResidualEvaluator.of(SPEC, Expressions.equal("id", 1), true);
+    FileScanTask fileScanTask =
+        new BaseFileScanTask(
+            FILE_A,
+            new DeleteFile[] {FILE_A_DELETES},
+            SchemaParser.toJson(SCHEMA),
+            PartitionSpecParser.toJson(SPEC),
+            residualEvaluator);
+
+    FetchScanTasksResponse response =
+        FetchScanTasksResponse.builder()
+            .withFileScanTasks(List.of(fileScanTask))
+            .withDeleteFiles(List.of(FILE_A_DELETES))
+            // assume you have set this already
```

Comment:
Same as below, we generally don't use second person in comments, let's remove/reword instances of those

Reply:
will remove